src/sys/socket/mod.rs

```diff
@@ -139,7 +139,7 @@ impl<'a> Iterator for CmsgIterator<'a> {
         if buf.len() < sizeof_cmsghdr {
             return None;
         }
-        let cmsg: &cmsghdr = unsafe { mem::transmute(buf.as_ptr()) };
+        let cmsg: &'a cmsghdr = unsafe { &*(buf.as_ptr() as *const cmsghdr) };
```

Comment:
To dereference the raw pointer, then get a reference to the value behind (the dereference doesn't copy the value out)